hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -624,7 +625,12 @@ public ReferenceCounted<IOmMetadataReader> getActiveFsMetadataOrSnapshot(
     String[] keyParts = keyName.split(OM_KEY_PREFIX);
     if (isSnapshotKey(keyParts)) {
       String snapshotName = keyParts[1];
-
+      // Updating the volumeName & bucketName in case the bucket is a linked bucket. We need to do this before a
```

Comment:
The method above this "deleteKeysFromDelKeyTableInSnapshotScope" also needs correction. Or are we planning to handle garbage collection separately?

Reply:
We have changed the snapshot sym link bucket name to the actual bucket name. We don't have to do anything about garbage collection code.

Reply:
Ok you mean that the callers ensure that the method is passed the real bucket name. Makes sense in that case.